src/wat-writer.cc

```diff
@@ -850,6 +918,22 @@ void WatWriter::FlushExprTree(const ExprTree& expr_tree) {
       break;
     }
 
+    case ExprType::TryBlock: {
+      auto try_ = cast<TryExpr>(expr_tree.expr);
+      WritePuts("(", NextChar::None);
+      WriteBeginBlock(LabelType::Try, try_->block,
+                      Opcode::Try_Opcode.GetName());
+      WriteFoldedExprList(try_->block->first);
+      FlushExprTreeStack();
+      for (const Catch* catch_ : try_->catches) {
+        ExprTree tree(catch_);
+        expr_tree_stack_.emplace_back(tree);
```

Comment:
This still seems more complex than necessary; since we know that the expression tree stack is empty at the beginning of every catch, why push it on just to flush it? If we just write out the catch statements directly (as done in `FlushCatchTree`) then we don't need to modify the `ExprTree` struct to support both `Expr` and `Catch`.